assets/javascripts/initializers/discourse-reactions.js.es6

```diff
@@ -49,6 +50,41 @@ function initializeDiscourseReactions(api) {
       post
     });
   });
+
+  api.modifyClass("component:emoji-value-list", {
+    didReceiveAttrs() {
+      this._super(...arguments);
+
+      if (this.setting.setting !== "discourse_reactions_enabled_reactions") {
+        return;
+      }
+
+      let defaultValue = this.values.includes(
+        this.siteSettings.discourse_reactions_like_icon
+      );
+
+      if (!defaultValue) {
+        this.collection.unshift({
+          emojiUrl: emojiUrlFor(
+            this.siteSettings.discourse_reactions_like_icon
+          ),
+          isEditable: false,
+          isEditing: false,
+          showUpDownButtons: this.collection.length - 1 ? true : false,
+          value: this.siteSettings.discourse_reactions_like_icon
+        });
+      } else {
+        this.collection.every(emoji => {
```

Comment:
what is every doing here ? where do you use the result of this ?

why don't we just do this:

```
this.collection.forEach(emoji => {
  if (emoji.value === this.siteSettings.discourse_reactions_like_icon) {
    emoji.isEditable = false;
  }
});
```

Reply:
or maybe even better:

```
const mainEmoji = this.collection.findBy("value", this.siteSettings.discourse_reactions_like_icon);
if (mainEmoji) {
  mainEmoji.isEditable = false;
}
```

Reply:
Sure, ill use findBy here